json_gpt.py

```diff
@@ -0,0 +1,85 @@
+import pandas as pd
+import requests
+from openai import OpenAI
+from dotenv import load_dotenv
+from os import environ
+from pydantic import BaseModel
+from collections import defaultdict
+import os
+
+# Load environment variables
+load_dotenv()
+
+# Set up API keys
+GITHUB_API_KEY = environ.get("GITHUB_PAT_2")
+client = OpenAI()  # Automatically loads OPENAI_API_KEY from .env
+
+# Read the JSON file
+df_json = pd.read_json('formatted_pull_requests_int.json')
+pd.set_option('display.max_colwidth', None)
+
+# Define response model
+class CommentCategoriesExtraction(BaseModel):
+    categories: list[str]
+
+# Initialize totals dictionary
+category_totals = defaultdict(int)
+output_file = 'category_totals_json.csv'
+
+# If the CSV exists, load it to resume progress
+if os.path.exists(output_file):
+    existing_df = pd.read_csv(output_file)
+    category_totals.update(existing_df.set_index('Category')['Total Count'].to_dict())
+
+# Process PRs
+batch_size = 5
+
+for start_idx in range(0, len(df_json), batch_size):
+    for idx in range(start_idx, min(start_idx + batch_size, len(df_json))):
+        test = df_json.iloc[idx]
+        pr_number = test['pull_request_url'].split('/')[-1]
+
+        # Fetch issue comments
+        issue_comments_url = f"https://api.github.com/repos/home-assistant/core/issues/{pr_number}/comments"
+        response = requests.get(issue_comments_url, headers={"Authorization": f"Bearer {GITHUB_API_KEY}"})
+        issue_comment_data = response.json()
+
+        # Filter out bot comments
+        issue_comment_bodies = [comment['body'] for comment in issue_comment_data if comment['user']['type'] != 'Bot']
+
+        # Fetch review comments
+        review_comments_url = f"https://api.github.com/repos/home-assistant/core/pulls/{pr_number}/comments"
+        response = requests.get(review_comments_url, headers={"Authorization": f"Bearer {GITHUB_API_KEY}"})
+        review_comment_data = response.json()
+
+        # Extract review comment bodies
+        review_comment_bodies = [comment['body'] for comment in review_comment_data]
+
+        # Prepare the input for the OpenAI API
+        body = f"ISSUE COMMENT BODIES: \n{issue_comment_bodies}\n\nREVIEW COMMENT BODIES: \n{review_comment_bodies}"
```

Comment:
Following on from another comment I left on other file, I think passing in the interleaved comment sequence is an improvement on just all of one followed by all of the other